### DISCUSSION TITLE
API for Uploading User Profile Images to Blob Storage

### DISCUSSION THREAD

**Discussion by ManojNathIC** [Sunbird-Ed/Community#728] [General Discussion]
Hi team,
 
@vinukumar-vs ,@rajeevsathish, @AmiableAnil , @Krishnaj20, @chitranshu-keshav, @SanthoshVasabhaktula, @gandham-santhosh,

Hope you are doing well!

**Site Environment Details**: SunbirdEd portal, version 4.10.0

We have a requirement to upload user profile images. Is there any API available that can upload images to a blob storage container and return the URL of the uploaded image?

CC: @komalm @manishakapadnis @paritshivani

**Comment by ManojNathIC** [comment]
Hi team,

Any update on this?
We are waiting for your valuable insights.

**Comment by amitpriyadarshi** [comment]
@ManojNathIC at user-service layer there is no api to upload user profile image, but you can upload users image as asset and use the url.
You can call content create and content upload api to upload the image. It will return you a asset id and its url.

> curl 'https://{{host}}/api/content/v1/create' \
  -H 'Authorization;' \
  -H 'Content-Type: application/json' \
  --data-raw '{"request":{"asset":{"primaryCategory":"asset","language":["English"],"code":"14847a49-43d5-4af3-9421-057a51baa0c1","name":"Screenshot 2024-05-29 at 3.25.24 PM","mediaType":"image","mimeType":"image/png","createdBy":"d416c869-27b3-491c-9562-c43ac04406df","creator":"apcontentcreator100","channel":"01269934121990553633","keywords":""}}}'


> curl 'https://{{host}}/api/content/v1/upload/do_21408586760487731212?enctype=multipart/form-data&processData=false&contentType=false&cache=false' \
  -H 'Authorization;' \
  -H 'Content-Type: multipart/form-data; boundary=----WebKitFormBoundarypM4Y6vxN2rqUTaql' \
  --data-raw $'------WebKitFormBoundarypM4Y6vxN2rqUTaql\r\nContent-Disposition: form-data; name="fileUrl"\r\n\r\nhttps://sunbirdstagingpublic.blob.core.windows.net/sunbird-content-staging/content/assets/do_21408586760487731212/screenshot-2024-05-29-at-3.25.24-pm.png\r\n------WebKitFormBoundarypM4Y6vxN2rqUTaql\r\nContent-Disposition: form-data; name="mimeType"\r\n\r\nimage/png\r\n------WebKitFormBoundarypM4Y6vxN2rqUTaql--\r\n'

NOTE: This will be uploaded as asset.
cc: @vinukumar-vs

**Comment by ManojNathIC** [comment]
@amitpriyadarshi  Thank you for your response,

Can we use the same API to upload user profile images in the different storage containers? 

_For example:_

 Now courses and content images are uploaded in the **content** container, And we need to upload user profile image in another container **profile-images**, because if we upload user profile images in **content** container they will appear while  uploading/creating images to the course/content

**Reply by amitpriyadarshi** [reply]
@ManojNathIC storage container value is configuration driven. If you update this value all the contents asset will also start moving to that container.

**Comment by HarishGangula** [comment]
As we haven't received a response from you, we will proceed to close this thread. If you have any further questions or require assistance in the future, please don't hesitate to start a new thread. Thank you.
